src/app/settings/observers.ts

```diff
@@ -11,3 +14,16 @@ export function watchLanguageChanges() {
     }
   });
 }
+
+export function watchLanguageChanges2(store: Store<RootState, any>) {
+  store.dispatch(
+    observe({
+      id: 'lang-observer',
+      getObserved: languageSelector,
+      sideEffect: ({ current }) => {
+        localStorage.setItem('dimLanguage', current);
+        i18next.changeLanguage(current);
+      },
+    }),
+  );
+}
```

Comment:
Just put it in this file so comparison is easy for the draft. I think I am likely to delete the file altogether and just dispatch the action directly in `index.tsx`

Reply:
Yeah you wouldn't want to grab the store reference here.